fe/fe-core/src/main/java/com/starrocks/catalog/FsBroker.java

```diff
@@ -89,10 +90,18 @@ public boolean handleHbResponse(BrokerHbResponse hbResponse) {
             // this heartbeat info also need to be synced to follower.
             // Since the failed heartbeat info also modifies fe's memory, (this.heartbeatRetryTimes++;)
             // if this heartbeat is not synchronized to the follower, 
-            // that will cause the Follower and master’s memory to be inconsistent
+            // that will cause the Follower and leader’s memory to be inconsistent
             isChanged = true;
         }
-
+        if (!isReplay) {
+            hbResponse.aliveStatus = isAlive ?
+                HeartbeatResponse.AliveStatus.ALIVE : HeartbeatResponse.AliveStatus.NOT_ALIVE;
+        } else {
+            if (hbResponse.aliveStatus != null) {
```

Comment:
Better add some comment that why hbResponse.aliveStatus can be null.

Reply:
done